TLM/TLM/Lifecycle/Patcher.cs

```diff
@@ -110,6 +112,46 @@ public static class Patcher {
             }
         }
 
+        /// <summary>
+        /// manually applying harmony patches if target mods are enabled
+        /// </summary>
+        /// <returns>false if patching failed, true otherwise</returns>
+        private static bool PatchManual(string harmonyId) {
+            try {
+                var harmony = new Harmony(harmonyId);
+
+                // Patching SimulationStepPatch1 in Reversible Tram AI mod
+                if (IsAssemblyEnabled("ReversibleTramAI")) {
```

Comment:
What happens if the assembly is enabled _after_ this code runs.

For example, user loads game to main menu, then in Steam they subscribe Reversible Tram AI mod (or they might already have it subscribed, but disabled) , then they enable it....?

I suspect we'll need to listen for plugin manager events (hooked via `TMPELifecycle`) to invoke `PatchManual()` whenever a mod is enabled/disabled, and in turn `PatchManual()` will need to test if it's already run, or something along those lines?

Reply:
Additionally, once patched, what happens if the user disables Reversible Tram AI mod? Do we need to also remove patches that TM:PE added?

Reply:
I am not sure how TMPE uses the Patcher class. My mods usually apply manual patches in OnLevelLoaded(). If users disable the reversible tram ai mod, Harmony patches in the ai mod will be unpatched. The patch in this PR is a patch for another patch, so I think TMPE doesn't need to remove this patch if the reversible tram ai mod is disabled

Reply:
> The patch in this PR is a patch for another patch, so I think TMPE doesn't need to remove this patch if the reversible tram ai mod is disabled

What if user disables TM:PE after it's patched your patch? Example: User is trying to track down a crash or some other bug, suspects it's TM:PE so disables TM:PE (which has already patched your patch) then loads city.

Reply:
@krzychu124 any thoughts on this? Is it something that needs to go in this PR or can it be added at later date? It would be good to get this in for 11.6.5 release if possible, assuming it's going to TEST workshop page first as that way we can start getting feedback from users.

Reply:
> What if user disables TM:PE after it's patched your patch? Example: User is trying to track down a crash or some other bug, suspects it's TM:PE so disables TM:PE (which has already patched your patch) then loads city.

I will test this later

Reply:
Lifecycle of patches was completely redesigned recently.

Patcher runs first time (only Pathfinding patches) when you click `Load Game` or start `New game` (level preload), because game is "warming up" pathfinding to fix paths of loaded public transport lines and `OnLevelLoaded` is way too late to do so - you would see disconnected lines after load, then runs again but this time with all remaining patches at the end of loading process (`OnLevelLoaded`).

> What if user disables TM:PE after it's patched your patch?

Obviously all TMPE patches are reverted (unpatched) when you disable it, no issue here.

> Additionally, once patched, what happens if the user disables Reversible Tram AI mod? Do we need to also remove patches that TM:PE added?

Same here, we are patching "Reversible Tram AI" - it's AI that drives trams -> no mod, no AI (vehicles should despawn but may throw error when assigned to line (not tested that)) - nothing will trigger that patch.


IMO it's ok, code looks good. I've tested it multiple times + few hot-reloads. Didn't notice any issues besides some weird situations which problably require a lot more complex code to solve (arriving trams from both sides of tram stop segment sometimes cause derail if connecting segments are not straight XD) 
We can release it in 11.6.5, I don't see anything that is holding it longer.

Reply:
Yes there may still be some weird edge cases like the one you mentioned. And I'm a bit worried that users might mistakenly think it's caused by TMPE and bother the TMPE team